stage #1/module #6. IO, NIO/lesson #1. IO/content.md

```diff
@@ -0,0 +1,124 @@
+# Java I/O (java.io)
+
+## Introduction to IO
+
+Java I/O (Input and Output) is used to process the input and produce the output.
+Java uses the concept of a stream to make I/O operation fast. The `java.io` package contains all the classes 
+required for input and output operations. 
+
+We can perform **file handling in Java** by Java I/O API.
+
+## Java I/O Streams. What is a Stream?
+
+A **stream** is an ordered sequence of bytes of indeterminate length. 
+A stream can represent many kinds of sources and destinations, including disk files, devices, other programs, and memory arrays. 
+In other words, we can imagine streams as water pipelines, data as water, source/target as well/end-user.
+
+Java I/O Streams support many kinds of data, including simple bytes, primitive data types, localized characters, and objects. Some streams simply pass on data;
+others manipulate and transform the data in useful ways
+
+- [Byte Streams](https://docs.oracle.com/javase/tutorial/essential/io/bytestreams.html) handle I/O of raw binary data.
+- [Character Streams](https://docs.oracle.com/javase/tutorial/essential/io/charstreams.html) handle I/O of character data, automatically handling translation to and from the local character set.
+- [Buffered Streams](https://docs.oracle.com/javase/tutorial/essential/io/buffers.html) optimize input and output by reducing the number of calls to the native API.
+- [Scanning and Formatting](https://docs.oracle.com/javase/tutorial/essential/io/scanfor.html) allows a program to read and write formatted text.
+- [I/O from the Command Line](https://docs.oracle.com/javase/tutorial/essential/io/cl.html) describes the Standard Streams and the Console object.
+- [Data Streams](https://docs.oracle.com/javase/tutorial/essential/io/datastreams.html) handle binary I/O of primitive data type and String values.
+- [Object Streams](https://docs.oracle.com/javase/tutorial/essential/io/objectstreams.html) handle binary I/O of objects.
+
+No matter how they work internally, all streams present the same simple model to programs that use them: A stream is a sequence of data. A program uses an *input stream* to read data from a source, one item at a time:
+![reading_stream_scheme.png](assets/reading_stream_scheme.png)
```

Comment:
Could you please add empty line before images?

Reply:
Added